### PR TITLE
BFD-2013: Namespace Cloudwatch metrics & add additional dimensions for agent to collect

### PR DESCRIPTION
**Pull request by kcirtapfromspace**
**JIRA Ticket:**
[BFD-2013](https://jira.cms.gov/browse/BFD-2013)

**User Story or Bug Summary:**

As an Engineer on the BFD team, I would like create dashboard views based on metrics collected by AWS Cloudwatch agent. In order to easily facilitate this namespace metrics & ensure there is a full suite of collected measurements.
---

### What Does This PR Do?

This PR adds metrics collected to the base configuration for the cloudwatch agent. Namespacing the metrics is added to specific service configuration which updates the agent during ansible run time. 
### What Should Reviewers Watch For?

If you're reviewing this PR, please check for these things in particular:

* Verify all PR security questions and checklists have been completed and addressed.

### What Security Implications Does This PR Have?

Submitters should complete the following questionnaire:

* If the answer to any of the questions below is **Yes**, then **you must** supply a link to the associated Security Impact Assessment (SIA), security checklist, or other similar document in Confluence here: **N/A**

    * Does this PR add any new software dependencies? 
      * [ ] Yes
      * [x] No
    * Does this PR modify or invalidate any of our security controls?
      * [ ] Yes
      * [x] No
    * Does this PR store or transmit data that was not stored or transmitted before?
      * [ ] Yes
      * [x] No

* If the answer to any of the questions below is **Yes**, then please add @StewGoin as a reviewer, and note that this PR **should not be merged** unless/until he also approves it.
    * Do you think this PR requires additional review of its security implications for other reasons?
      * [ ] Yes
      * [x] No

### What Needs to Be Merged and Deployed Before this PR?

This PR cannot be either merged or deployed until the following prerequisite changes have been fully deployed:

* N/A

### Submitter Checklist

I have gone through and verified that...:

* [x] I have named this PR and branch so they are [automatically linked](https://confluence.atlassian.com/adminjiracloud/integrating-with-development-tools-776636216.html) to the (most) relevant Jira issue. Ie: `BFD-123: Adds foo`
* [x] This PR is reasonably limited in scope, to help ensure that:
    1. It doesn't unnecessarily tie a bunch of disparate features, fixes, refactorings, etc. together.
    2. There isn't too much of a burden on reviewers.
    3. Any problems it causes have a small "blast radius".
    4. It'll be easier to rollback if that becomes necessary.
* [x] This PR includes any required documentation changes, including `README` updates and changelog / release notes entries.
* [x] All new and modified code is appropriately commented, such that the what and why of its design would be reasonably clear to engineers, preferably ones unfamiliar with the project.
* [x] All tech debt and/or shortcomings introduced by this PR are detailed in `TODO` and/or `FIXME` comments, which include a JIRA ticket ID for any items that require urgent attention.
* [x] Reviews are requested from both:
    * At least two other engineers on this project, at least one of whom is a senior engineer or owns the relevant component(s) here.
    * Any relevant engineers on other projects (e.g. DC GEO, BB2, etc.).
* [x] Any deviations from the other policies in the [DASG Engineering Standards](https://github.com/CMSgov/cms-oeda-dasg/blob/master/policies/engineering_standards.md) are specifically called out in this PR, above.
    * Please review the standards every few months to ensure you're familiar with them.

### CODE REVIEW

File: ops/ansible/roles/cloudwatch-agent-instrumented/templates/cwagent.json.j2 [hunk 1]
```diff
@@ -12,27 +12,64 @@
           "usage_idle",
           "usage_iowait",
           "usage_user",
-          "usage_system"
+          "usage_system",
+          "usage_active" 
+
```
Comment (mjburling) [inline ops/ansible/roles/cloudwatch-agent-instrumented/templates/cwagent.json.j2:17]: Seems like an extraneous blank line here.
```suggestion
```

**Review by mjburling — CHANGES_REQUESTED** [review]
Namespaces ought to help along with a pretty reasonable set of additional metrics, too!

I have two notes on formatting–in addition to the one in-line comment, I wonder what you would think about sorting the measurement arrays? It's not critical, but I can appreciate lexicographic ordering every now and again. It could make sense here.

File: ops/ansible/roles/cloudwatch-agent-instrumented/templates/cwagent.json.j2 (hunk 1 above)
Comment (kcirtapfromspace) [reply ops/ansible/roles/cloudwatch-agent-instrumented/templates/cwagent.json.j2:17]: I'll strike the line

**Review by mjburling — APPROVED** [review]
Looks great! Thanks for looking into sorting, etc.

While taking a final look, I remembered something I meant to bring up in a previous discussion–I haven't got experience with process monitoring in CloudWatch, but I've used competing products that offer the same feature. In previous roles, it was widely regarded as cost-prohibitive to toggle process monitoring in the general case, so we were asked to only activate as needed and even then, to employ judicious filtering on known processes. Do you have any numbers that might support what the cost impact might be, even for the humble ~18 instances we generally support?

I'm approving just the same, but we'll need to keep an eye out for any marked upticks in our spending in the event that it is as costly.

**Comment by kcirtapfromspace** [comment]
Good call out on thinking about costs. I'm no entirely sure if these are considered custom metrics or inherent to detailed metrics, but I'll make the assumption that they are custom metrics.  That would mostly explains why I'm having trouble being able correlate cost back to the test instance being used, as I establish that its costs $0.03 per day in the billing console. 

AWS charges on a custom metrics count per month. There are currently 11198 defined metrics in the CWAgent namespace(My assumptions based on [pricing examples](https://aws.amazon.com/cloudwatch/pricing/) show costs should be about ~$4500 per month), my napkin math would bring this up to 14472 incurring roughly a ~$327 additional charge. Additionally the collection of metric Monthly API requests totals tally up to being well below (best guestimate is 200k) the 1M threshold to incur fees .
Attempts to actually correlate this math to the billing console are fraught as AWS billing costs show total ownership costs for cloudwatch under $3k/month for the past 3 months
image

**Review by keithdadkins — APPROVED** [review]

**Pull request merged**